crates/eww/src/app.rs

```diff
@@ -362,29 +363,35 @@ fn apply_window_position(
     monitor_geometry: gdk::Rectangle,
     window: &gtk::Window,
 ) -> Result<()> {
-    let gdk_window = window.get_window().context("Failed to get gdk window from gtk window")?;
-    window_geometry.size = Coords::from_pixels(window.get_size());
+    let gdk_window = window.window().context("Failed to get gdk window from gtk window")?;
+    window_geometry.size = Coords::from_pixels(window.size());
     let actual_window_rect = get_window_rectangle(window_geometry, monitor_geometry);
-    gdk_window.move_(actual_window_rect.x, actual_window_rect.y);
+
+    let size = gdk_window.origin();
+
+    if actual_window_rect.x != size.1 || actual_window_rect.y != size.2 {
+        gdk_window.move_(actual_window_rect.x, actual_window_rect.y);
+    }
```

Comment:
`origin` does not return the size, but the position of the window
comparing the position to the size would also not make sense - the naming here is thus _really_ weird.

Reply:
true, I missed renaming that, thanks for the heads up :sweat_smile: